packages/flutter_migrate/lib/src/compute.dart

```diff
@@ -0,0 +1,1014 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'base/common.dart';
+import 'base/file_system.dart';
+import 'base/logger.dart';
+import 'base/project.dart';
+import 'base/terminal.dart';
+import 'custom_merge.dart';
+import 'environment.dart';
+import 'flutter_project_metadata.dart';
+import 'result.dart';
+import 'utils.dart';
+
+// This defines files and directories that should be skipped regardless
+// of gitignore and config settings. Using `/` as a stand-in for path separator.
+const List<String> _skippedFiles = <String>[
+  'ios/Runner.xcodeproj/project.pbxproj', // Xcode managed configs that may not merge cleanly.
+  'README.md', // changes to this shouldn't be overwritten since is is user owned.
+];
+
+const List<String> _skippedDirectories = <String>[
+  '.dart_tool', // ignore the .dart_tool generated dir
+  '.git', // ignore the git metadata
+  'lib', // Files here are always user owned and we don't want to overwrite their apps.
+  'test', // Files here are typically user owned and flutter-side changes are not relevant.
+  'assets', // Common directory for user assets.
+  'build' // Build artifacts
+];
+
+bool _skipped(String localPath, FileSystem fileSystem,
```

Comment:
This could use docs about what the prefixes should be (relative path prefixes? component prefixes? basename prefixes?)